### PR TITLE
fix: login / signin frontend router

### PR DESCRIPTION
**Pull request by imchell**
Fix: #1227 

substitute goToLogin with frontend router

login

### CODE REVIEW

**Comment by hsluoyz** [comment]
@leo220yuyaodog plz review

File: web/src/auth/LoginPage.js [hunk 1]
```diff
@@ -302,13 +302,9 @@ class LoginPage extends React.Component {
...
           subTitle={"The application does not allow to sign up new account"}
           extra={[
-            <Link key="login" onClick={() => {
-              Setting.goToLogin(this, application);
-            }}>
-              <Button type="primary" key="signin">
-                Sign In
-              </Button>
-            </Link>,
+            <Button type="primary" key="signin">
```
Comment (hsluoyz) [inline web/src/auth/LoginPage.js:305]: Link inside Button? Does the UI work right?

File: web/src/auth/LoginPage.js [hunk 2]
```diff
@@ -468,11 +464,7 @@ class LoginPage extends React.Component {
       return (
         <div style={{float: "right"}}>
           {i18next.t("signup:Have account?")}&nbsp;
-          <Link onClick={() => {
-            Setting.goToLogin(this, application);
-          }}>
-            {i18next.t("signup:sign in now")}
-          </Link>
+          {renderLoginLink(application, i18next.t("signup:sign in now"))}
```
Comment (hsluoyz) [inline web/src/auth/LoginPage.js:467]: {
  xxx
}

File: web/src/auth/PromptPage.js [hunk 3]
```diff
@@ -22,6 +22,7 @@ import i18next from "i18next";
 import AffiliationSelect from "../common/AffiliationSelect";
 import OAuthWidget from "../common/OAuthWidget";
 import SelectRegionBox from "../SelectRegionBox";
+import {redirectToLoginPage, renderLoginLink} from "../Setting";
```
Comment (hsluoyz) [inline web/src/auth/PromptPage.js:25]: Remove this

File: web/src/auth/PromptPage.js [hunk 4]
```diff
@@ -190,7 +191,7 @@ class PromptPage extends React.Component {
           if (redirectUrl !== "" && redirectUrl !== null) {
             Setting.goToLink(redirectUrl);
           } else {
-            Setting.goToLogin(this, this.getApplicationObj());
+            redirectToLoginPage(this.getApplicationObj());
```
Comment (hsluoyz) [inline web/src/auth/PromptPage.js:194]: Setting.redirectToLoginPage

**Review by hsluoyz — DISMISSED** [review]

File: web/src/auth/LoginPage.js (hunk 1 above)
Comment (leo220yuyaodog) [reply web/src/auth/LoginPage.js:305]: UI works right.

**Comment by leo220yuyaodog** [comment]
UI works right.

**Pull request merged**